### DISCUSSION TITLE
Launching Discussions for TorchSharp.PyBridge: Open Collaboration and Upcoming Features

### DISCUSSION THREAD

**Discussion by shaltielshmid** [shaltielshmid/TorchSharp.PyBridge#3] [Announcements]
Hello TorchSharp.PyBridge enthusiasts!

We are thrilled to open up discussions for our project, aiming to foster collaboration and innovation within our community. This platform will be your go-to place for sharing ideas, asking questions, and getting involved in the ongoing development of TorchSharp.PyBridge.

One of our key goals is to enhance the seamless interoperability between .NET and Python, particularly in model serialization. While we've made significant strides with features like load_py and save_py for Modules, we're continuously working to expand and improve our capabilities.

A noteworthy point on our roadmap is the development of support for Optimizer state_dict. This feature is currently in the works, and we're excited about the potential it holds for our users. Your feedback and contributions will be invaluable in shaping this and other features.

We invite you to share your thoughts, suggest improvements, and possibly contribute code or ideas. Whether you're a seasoned developer or just starting, your input is crucial for the growth of this project.

Looking forward to your active participation and innovative contributions!

**Comment by GeorgeS2019** [comment]
https://github.com/dotnet/TorchSharp/pull/1138#issuecomment-1793811353

Is it possible to this as a future feature?

# Option ONE
It will reconstruct the PyTorch model in **TorchSharp** c# codes given the information of 
- the PyTorch Module and 
- the saved state_dict

# Option TWO
https://github.com/dotnet/TorchSharp/pull/1138#issuecomment-1793790853
Given the ONNX exported out of the PyTorch Model, It will reconstruct the PyTorch model in **TorchSharp** c# codes given the information of 
- exported ONNX in a similar way as in [onnx2torch](https://github.com/dotnet/TorchSharp/discussions/1111)


#### Additional Ref
https://github.com/fel88/Dendrite/issues/11
